unittests/Feature/FeatureModelParser.cpp

```diff
@@ -53,6 +53,75 @@ TEST(FeatureModelParser, outOfOrder) {
   EXPECT_TRUE(P.buildFeatureModel());
 }
 
+// TODO: Detect error while parsing
+//
+//  TEST(FeatureModelParser, errorMismatchParentChild) {
+//   auto FS = llvm::MemoryBuffer::getFileAsStream(
+//       getTestResource("error_mismatch_parent_child.xml"));
+//
+//   ASSERT_TRUE(FS);
+//
+//   auto P = FeatureModelXmlParser(FS.get()->getBuffer().str());
+//
+//   EXPECT_TRUE(P.verifyFeatureModel());
+//   EXPECT_FALSE(P.buildFeatureModel());
+// }
+
+TEST(FeatureModelParser, errorMissingChild) {
+  auto FS = llvm::MemoryBuffer::getFileAsStream(
+      getTestResource("error_missing_child.xml"));
+  ASSERT_TRUE(FS);
+
+  auto P = FeatureModelXmlParser(FS.get()->getBuffer().str());
+
+  EXPECT_TRUE(P.verifyFeatureModel());
+  EXPECT_FALSE(P.buildFeatureModel());
```

Comment:
To simplify things, we should move this into a method that does the initial checks and returns P, should some test want to test extra stuff. With this, a test is only putting a string into the new method.